### PR TITLE
Fixed some bug in multi source lensing and los.

### PR DESCRIPTION
**Pull request by nkhadka21**

### CODE REVIEW

File: slsim/lens.py [hunk 1]
```diff
@@ -376,7 +376,7 @@ def los_linear_distortions(self):
         """
         kappa = self.los_class.convergence
         gamma1, gamma2 = self.los_class.shear
-        return kappa, gamma1, gamma2
+        return gamma1, gamma2, kappa
```
Comment (sibirrer) [inline slsim/lens.py:379]: I'd like to have it defined the opposite way with kappa first (see also the docstrings in this routine

File: slsim/lens.py [hunk 2]
```diff
@@ -355,7 +355,7 @@ def deflector_redshift(self):
 
         :return: lens redshift
         """
-        return self.deflector.redshift
+        return float(self.deflector.redshift)
```
Comment (sibirrer) [inline slsim/lens.py:358]: I already made changes in my PR for this. I'd like to have the self.deflector.redshift quantity well defined as float, not the return of the function in lens.py (same for the source redshifts as well

File: slsim/lens.py [hunk 3]
```diff
@@ -732,6 +732,12 @@ def _extended_single_source_magnification(self, source, source_index):
                 source_index
             ]
             kwargs_source_mag = kwargs_params["kwargs_source"][source_index]
+        if isinstance(light_model_list, list):
```
Comment (sibirrer) [inline slsim/lens.py:735]: I don't think we need those things as the components should already be defined as lists

**Review by sibirrer — CHANGES_REQUESTED** [review]
Thank you @nkhadka21. I hope you saw my PR that addresses most of these things. I have the tests working and have also some more changes implemented (including the minor bug in the lens equation solver etc)

File: slsim/lens.py (hunk 1 above)
Comment (nkhadka21) [reply slsim/lens.py:379]: Yes, I saw your PR! and changing things occordingly.

File: slsim/lens.py (hunk 3 above)
Comment (nkhadka21) [reply slsim/lens.py:735]: These lines are applicable while accessing keywords in some case (just one case where we have multisource and single sersic light profile. In this case above line will produce str and dictionary due to the indexing. So, need to convert them to list.)

**Pull request closed without merging**